pyanaconda/payload/flatpak.py

```diff
@@ -0,0 +1,353 @@
+
+    def install_all(self):
+        """Install all the refs contained on the remote."""
+        self._stuff_refs_to_transaction()
+
+        try:
+            self._transaction.run()
+        except GError as exn:
+            raise FlatpakInstallError(exn.message)
+
+    def _stuff_refs_to_transaction(self):
+        for ref in self._remote_refs_list.get_refs_full_format():
+            self._transaction.add_install(self.LOCAL_REMOTE_NAME, ref, None)
+
+    def replace_installed_refs_remote(self, new_remote):
+        """Replace remote on all the installed refs.
+
+        :param str new_remote: name of the new remote
+        """
+        installed_refs_list = InstalledRefsList(self._transaction.get_installation())
+
+        installed_refs_list.replace_installed_refs_remote(new_remote)
+
+    def _operation_started_callback(self, transaction, operation, progress):
+        """Start of the new operation.
+
+        :param transaction: the main transaction object
+        :type transaction: Flatpak.Transaction instance
+        :param operation: object describing the operation
+        :type operation: Flatpak.TransactionOperation instance
+        :param progress: object providing progess of the operation
+        :type progress: Flatpak.TransactionProgress instance
+        """
+        self._log_operation(operation, "started")
+        progressQ.send_message(_("Installing %(flatpak_name)s") %
+                               {"flatpak_name": operation.get_ref()})
+
+    def _operation_stopped_callback(self, transaction, operation, commit, result):
+        """Existing operation ended.
+
+        :param transaction: the main transaction object
+        :type transaction: Flatpak.Transaction instance
+        :param operation: object describing the operation
+        :type operation: Flatpak.TransactionOperation instance
+        :param str commit: operation was committed this is a commit id
+        :param result: object containing details about the result of the operation
+        :type result: Flatpak.TransactionResult instance
+        """
+        self._log_operation(operation, "stopped")
+
+    def _operation_error_callback(self, transaction, operation, error, details):
+        """Process error raised by the flatpak operation.
+
+        :param transaction: the main transaction object
+        :type transaction: Flatpak.Transaction instance
+        :param operation: object describing the operation
+        :type operation: Flatpak.TransactionOperation instance
+        :param error: object containing error description
+        :type error: GLib.Error instance
+        :param details: information if the error was fatal
+        :type details: int value of Flatpak.TransactionErrorDetails
+        """
+        self._log_operation(operation, "failed")
+        log.error("Flatpak installation failed with message: '%s' -- error is fatal %s",
+                  error.message, self._error_fatal)
```

Comment:
Is there _error_fatal attribute in the instance?

Reply:
A valid bug found also by pylint. I should have run the pylint before posting PR.

Reply:
Fixed.